module/spl/spl-kmem-cache.c

```diff
@@ -1001,8 +1003,17 @@ spl_kmem_cache_create(char *name, size_t size, size_t align,
 		slabflags |= SLAB_USERCOPY;
 #endif
 
-		skc->skc_linux_cache = kmem_cache_create(
-		    skc->skc_name, size, align, slabflags, NULL);
+#if defined(KMEM_CACHE_USERCOPY)
+        /*
+         * Newer grsec patchset uses kmem_cache_create_usercopy()
+         * instead of SLAB_USERCOPY flag
+         */
+        skc->skc_linux_cache = kmem_cache_create_usercopy(
```

Comment:
Rather than relying on KMEM_CACHE_USERCOPY being defined to imply `kmem_cache_create_usercopy()` exists a configure check should be added.  You can model the new check on the existing SPL_AC_FS_STRUCT_SPINLOCK check in `spl-build.m4`.  It's important to call the new function with local variables which match the types used by the target function.  This way API changes can be caught at configure time.